[DOCS-6434] Add Log Pipeline Scanner doc (#20594)
DataDog/documentation: content/en/logs/log_configuration/pipeline_scanner.md

---
title: Pipeline Scanner
kind: documentation
disable_toc: false
further_reading:
    - link: "/logs/log_configuration/pipelines/"
      tag: "Documentation"
      text: "Learn more about log pipelines"
    - link: "/logs/log_configuration/processors/"
      tag: "Documentation"
      text: "Learn more about log processors"
---

> <div class="alert alert-warning">Log Pipelines Scanner is in beta. </div>

Author: @LucasC happy to update this, but can you add to this [confluence doc](https://datadoghq.atlassian.net/wiki/spaces/TS/pages/328404964/The+Public+Docs+say+Contact+Support) what Support should do when someone requests access?

> <div class="alert alert-warning">Log Pipelines Scanner is in beta. </div>

Reviewer: Thank you, I wasn't aware of this page. Added an entry [here](https://datadoghq.atlassian.net/wiki/spaces/TS/pages/328404964/The+Public+Docs+say+Contact+Support#Log-Management---Log-Pipelines-Scanner-Private-beta-access), as well as documented the process [there](https://datadoghq.atlassian.net/wiki/spaces/TS/pages/3292858144/Providing+access+to+Log+Pipelines+Scanner+Private+Beta).

> <div class="alert alert-warning">Log Pipelines Scanner is in beta. </div>

Author: Perfect, thanks!

## Overview

> The Log Pipeline Scanner helps you to inspect your logs so that you can find out which log pipelines and processors are applied to them. Log pipelines are made up of a sequential chain of processors that parse and enrich your logs. A large organization might have hundreds of pipelines that modify logs from multiple services. These pipelines are set up and managed by different teams for their specific use cases, such as security monitoring, compliance audits, and DevOps. If you are trying to parse new logs, it might be difficult to know which processors and pipelines are modifying the logs.

Reviewer: ```suggestion
The Log Pipelines Scanner allows you to scan log pipelines in real time, tracing specific logs and quickly identifying which pipelines and processing rules have made changes to its fields. Organizations rely on log pipelines to process extensive log volumes, each team restructuring and enriching logs differently for their specific use cases, such as security monitoring, compliance audits, and DevOps.
```

> Log Pipeline Scanner traces your queried logs so that you can see in real time which pipelines and processing rules have modified them. This lets you and your team:

Reviewer: ```suggestion
This lets you and your team:
```

- Troubleshoot log processing issues, such as unparsed logs, missing tags, or unexpected changes to the log structure.
- Determine and remove conflicting or redundant processing rules.
- Ensure that logs going through log pipelines are meeting security and compliance requirements.

## Identify pipelines and processors modifying a log

The Pipeline Scanner samples and annotates logs matching the search query with the different processing steps they are going through, making it possible to identify all changes to your logs.

1. Navigate to [Log Explorer][1].
1. Click on a log where you want to find out which pipelines and processors are modifying it.
1. Click on the three dots in the upper right corner of the panel.
1. Select **Live pipeline debugging**.   
    Alternatively, click on an attribute in the log panel and select **Pipeline Scanner**. 

> 1. You can further refine your query in the Pipeline Scanner page. This query cannot be changed once a session is started.

Reviewer 2: ```suggestion
1. You can further refine your query in the Pipeline Scanner page. This query cannot be changed once a session is started.
```

Is it possible yet to link to the in app Pipeline Scanner page ?

Reviewer: We will be able to. @JeanCoquelet to confirm the exact URL, but probably something like `app.datadoghq.com/logs/pipelines?query`

1. Click **Launch this session**.   
    For the next 15 minutes, logs matching your query are tagged with information about which pipelines and processors are modifying those logs. The Live Tail in the scanner shows which pipelines and how many pipelines match each of the logs.
1. Click a log to see the list of pipelines and processors matched to that log. Live Tail is paused at this point.

You can modify the pipelines and processors in the right side panel. The modifications you make do not affect the logs that have already been processed. Click **Play** to view the new logs that have been modified by the updated pipelines and processors.

You can also access Pipeline Scanner in the Log Pipelines page and then define the query for the logs you want to inspect.

### Limitations

> 

Reviewer: ```suggestion

The `logs_write_pipelines` permission is required to use the Pipeline Scanner.
```

Reviewer: + a link to https://docs.datadoghq.com/account_management/rbac/permissions/#log-management I guess :)

Author: link added!

> You can also access Pipeline Scanner in the Log Pipelines page and then define the query for the logs you want to inspect.

Reviewer 2: ```suggestion
You can also access Pipeline Scanner in the Log Pipelines page and then define the query for the logs you want to inspect.
```

Same with this one in that I think it would be helpful to link to the Log Pipelines page if possible :)

The number of sessions you can launch is limited to:

> - A maximum of 3 concurrent sessions

Reviewer 2: ```suggestion
- A maximum of 3 concurrent sessions.
```

> - A maximum of 12 sessions over a 12-hour sliding window

Reviewer 2: ```suggestion
- A maximum of 12 sessions over a 12-hour sliding window.
```

---
title: Pipeline Scanner
kind: documentation
disable_toc: false
further_reading:
    - link: "/logs/log_configuration/pipelines/"
      tag: "Documentation"
      text: "Learn more about log pipelines"
    - link: "/logs/log_configuration/processors/"
      tag: "Documentation"
      text: "Learn more about log processors"
---

<div class="alert alert-warning">Log Pipelines Scanner is in private beta. <a href="/help">Contact Support</a> to request access. </div>

## Overview

Log Pipelines Scanner allows you to scan log pipelines in real time, trace specific logs, and identify which pipelines and processing rules made changes to its fields. Organizations rely on log pipelines to process extensive log volumes, each team restructuring and enriching logs differently for their specific use cases, such as security monitoring, compliance audits, and DevOps.

Use Pipeline Scanner to:

- Troubleshoot log processing issues, such as unparsed logs, missing tags, or unexpected changes to the log structure.
- Determine and remove conflicting or redundant processing rules.
- Ensure that logs going through log pipelines are meeting security and compliance requirements.

> {{< img src="logs/log_configuration/pipeline_scanner/pipeline_scanner.png" alt="The Pipeline Scanner showing two logs matched to the query, log details for the selected log, and the two pipelines modifying the queried logs" style="width:80%;" >}}

Author: ```suggestion
{{< img src="logs/log_configuration/pipeline_scanner/pipeline_scanner.png" alt="The Pipeline Scanner showing two logs that match the query, log details for the selected log, and the two pipelines modifying the queried logs" style="width:80%;" >}}
```
